lib/spack/spack/binary_distribution.py

```diff
@@ -414,11 +414,10 @@ def _fetch_and_cache_index(self, mirror_url, expect_hash=None):
             _, _, fs = web_util.read_from_url(hash_fetch_url)
             fetched_hash = codecs.getreader('utf-8')(fs).read()
         except (URLError, web_util.SpackWebError) as url_err:
-            errors.append(
-                RuntimeError("Unable to read index hash {0} due to {1}: {2}".format(
-                    hash_fetch_url, url_err.__class__.__name__, str(url_err)
-                ))
-            )
+            # If we cannot fetch the hash, we will always fetch the index
+            # No need to error since it's just the hash
+            tty.debug("Cannot read index hash {0} due to {1}: {2}".format(
```

Comment:
This would be bad for bootstrap errors, please keep the list of error messages like they were.

You really want to keep track of all errors across all mirrors, and when none of them succeed, raise an exception with a list of all problems.

Reply:
https://github.com/spack/spack/pull/26599 see the "to reproduce this" and try for yourself

Reply:
It's not an error not to find any mirrors, it just means the mirrors don't have any specs.

The bootstrap code needs to raise an error if none of its methods succeed, but that's about bootstrapping, not about binary cache fetching. This is the wrong place for it, and having it is why we were seeing errors when we should not have.

Reply:
I'm not going to accept this PR as long as it's reverting my work on improving error bootstrapping errors. Just catch the exception and ignore it then in the other code path. `tty.debug` is a terrible way to silence these issues, nobody wants to run `spack -d ...` to figure out bootstrapping problems, and errors should appear when running spack in non-interactive mode.